osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -319,28 +284,80 @@ private void load(LocalisationManager localisation)
                     }
                 };
 
+                addInfoLabels();
+
                 titleBinding.BindValueChanged(_ => setMetadata(metadata.Source));
                 artistBinding.BindValueChanged(_ => setMetadata(metadata.Source), true);
+                starDifficulty.BindValueChanged(updateDifficulty, true);
 
                 // no difficulty means it can't have a status to show
                 if (beatmapInfo.Version == null)
                     StatusPill.Hide();
+            }
 
-                addInfoLabels();
+            private void updateDifficulty(ValueChangedEvent<StarDifficulty?> valueChanged)
+            {
+                var difficulty = valueChanged.NewValue ?? new StarDifficulty();
+
+                topRightMetadataContainer.Child.FadeOut(250);
+                topRightMetadataContainer.Child.Expire();
+                topRightMetadataContainer.Child = createTopRightMetadataContainer(beatmap.BeatmapInfo, difficulty);
+
+                difficultyColourBarContainer.Child.Expire();
+                difficultyColourBarContainer.Child = createDifficultyColourBar(difficulty);
             }
 
-            private static Drawable createStarRatingDisplay(StarDifficulty difficulty) => difficulty.Stars > 0
-                ? new StarRatingDisplay(difficulty)
+            private DifficultyColourBar createDifficultyColourBar(StarDifficulty difficulty)
+                => new DifficultyColourBar(difficulty)
+                {
+                    RelativeSizeAxes = Axes.Y,
+                    Width = 20,
+                };
+
+            private FillFlowContainer createTopRightMetadataContainer(BeatmapInfo beatmapInfo, StarDifficulty difficulty)
+            {
+                var container = new FillFlowContainer
+                {
+                    Direction = FillDirection.Vertical,
+                    AutoSizeAxes = Axes.Both,
+                };
+
+                if (difficulty.Stars > 0)
                 {
-                    Margin = new MarginPadding { Bottom = 5 }
+                    container.Add(new StarRatingDisplay(difficulty)
+                    {
+                        Anchor = Anchor.TopRight,
+                        Origin = Anchor.TopRight,
+                        Shear = -wedged_container_shear,
+                        Margin = new MarginPadding { Bottom = 5 }
+                    });
                 }
-                : Empty();
+
```

Comment:
can all this be restored? i can't read this.

Reply:
also we really don't need private functions to create single drawables.

Reply:
What do you want to have restored? The `StarRatingDisplay` itself is the same as before, it just needs to get recreated on a difficulty change, the same for `DifficultyColourBar` otherwise both won't change if you change mods. Before the whole wedge was recreated on a mod change. I could recreate the whole info layer on a mod change if you would prefer that.
I recreate the whole container which contains the `StarRatingDisplay` and the `StatusPill` to maintain the order. I could also wrap a container around the `StarRatingDisplay` and expire only the Child of that

I added a private function bc i create the `DifficultyColourBar` at different places. Maybe i can skip it in the `load` function and do a check if a child exists before `Expire`

Reply:
You're adding more complexity around the place. Everything should pass `StarDifficulty` as it was previously, not the `beatmapInfo`.